nifi-nar-bundles/nifi-standard-bundle/nifi-standard-processors/src/main/java/org/apache/nifi/processors/standard/AttributesToJSON.java

```diff
@@ -282,17 +295,21 @@ public void onTrigger(ProcessContext context, ProcessSession session) throws Pro
         final Map<String, Object> atrList = buildAttributesMapForFlowFile(original, attributes, attributesToRemove, nullValueForEmptyString, pattern);
 
         try {
-            Map<String, Object> formattedAttributes = getFormattedAttributes(atrList);
+            final Map<String, Object> formattedAttributes = getFormattedAttributes(atrList);
             if (destinationContent) {
                 FlowFile conFlowfile = session.write(original, (in, out) -> {
                     try (OutputStream outputStream = new BufferedOutputStream(out)) {
-                        outputStream.write(objectMapper.writeValueAsBytes(formattedAttributes));
+                        if(prettyPrint) {
+                            outputStream.write(OBJECT_MAPPER.writerWithDefaultPrettyPrinter().writeValueAsBytes(formattedAttributes));
```

Comment:
Rather than calling `writerWithDefaultPrettyPrinter()` for every invocation, it would be better to assign the appropriate `ObjectWriter()` and then the `onTrigger` method does not need to be conditional.